src/query/db_accessor.cpp

```diff
@@ -0,0 +1,155 @@
+  auto result = db_accessor_->RemoveEdge(edge);
+  if (result.HasError() || !*result) {
+    return result;
+  }
+  return this->graph_->RemoveEdge(*edge);
+}
+
+storage::Result<EdgeAccessor> SubgraphDbAccessor::InsertEdge(SubgraphVertexAccessor *from, SubgraphVertexAccessor *to,
+                                                             const storage::EdgeTypeId &edge_type) {
+  VertexAccessor *from_impl = &from->impl_;
+  VertexAccessor *to_impl = &to->impl_;
+
+  auto result = db_accessor_->InsertEdge(from_impl, to_impl, edge_type);
+  if (result.HasError()) {
+    return result;
+  }
+  this->graph_->InsertEdge(*result);
+  return result;
+}
+
+storage::Result<std::optional<std::pair<VertexAccessor, std::vector<EdgeAccessor>>>>
+SubgraphDbAccessor::DetachRemoveVertex(  // NOLINT(readability-convert-member-functions-to-static)
+    VertexAccessor *) {                  // NOLINT(hicpp-named-parameter)
+  throw std::logic_error{"Such operation not possible on subgraph"};
+}
+
+storage::Result<std::optional<VertexAccessor>> SubgraphDbAccessor::RemoveVertex(
+    SubgraphVertexAccessor *subgraphvertex_accessor) {
+  VertexAccessor *vertex_accessor = &subgraphvertex_accessor->impl_;
+  auto result = db_accessor_->RemoveVertex(vertex_accessor);
+  if (result.HasError() || !*result) {
+    return result;
+  }
+  return this->graph_->RemoveVertex(*vertex_accessor);
+}
+
+SubgraphVertexAccessor SubgraphDbAccessor::InsertVertex() {
+  VertexAccessor vertex = db_accessor_->InsertVertex();
+  this->graph_->InsertVertex(vertex);
+  return SubgraphVertexAccessor(vertex, this->getGraph());
+}
+
+VerticesIterable SubgraphDbAccessor::Vertices(storage::View) {  // NOLINT(hicpp-named-parameter)
+  return VerticesIterable(graph_->vertices());
+}
+
+std::optional<VertexAccessor> SubgraphDbAccessor::FindVertex(storage::Gid gid, storage::View view) {
+  std::optional<VertexAccessor> maybe_vertex = db_accessor_->FindVertex(gid, view);
+  if (maybe_vertex && this->graph_->ContainsVertex(*maybe_vertex)) {
+    return *maybe_vertex;
+  }
+  return std::nullopt;
+}
+
+query::Graph *SubgraphDbAccessor::getGraph() { return graph_; }
+
+VertexAccessor SubgraphVertexAccessor::GetVertexAccessor() const { return impl_; }
+
+auto SubgraphVertexAccessor::OutEdges(storage::View view) const -> decltype(impl_.OutEdges(view)) {
+  auto maybe_edges = impl_.impl_.OutEdges(view, {});
+  if (maybe_edges.HasError()) return maybe_edges.GetError();
+  auto edges = std::move(*maybe_edges);
+  auto graph_edges = graph_->edges();
+
+  std::unordered_set<storage::EdgeAccessor> graph_edges_storage;
+
+  for (auto e : graph_edges) {
+    graph_edges_storage.insert(e.impl_);
+  }
+
+  std::vector<storage::EdgeAccessor> filteredOutEdges;
+  for (auto &edge : edges) {
+    if (std::find(begin(graph_edges_storage), end(graph_edges_storage), edge) != std::end(graph_edges_storage)) {
+      filteredOutEdges.push_back(edge);
+    }
+  }
```

Comment:
I am not satisfied with this chunk, I can make it better by allowing a comparison between query::VertexAccessor and storage::VertexAccessor probably. What do you guys think?

Reply:
Use `std::transform` for this. What would be better if you had that comparison?

Reply:
It would be easier if I could write just this:
```
auto maybe_edges = impl_.impl_.OutEdges(view, {});
 if (maybe_edges.HasError()) return maybe_edges.GetError();
 auto edges = std::move(*maybe_edges);
 auto graph_edges = graph_->edges();
for (auto &edge : edges) {
    if (std::find(begin(graph_edges), end(graph_edges), edge) != std::end(graph_edges)) {
      filteredOutEdges.push_back(edge);
    }
  }
```
But comparison between storage::VertexAccessor and query::VertexAccessor is not supported.

Reply:
Fixed it as we talked about on meet - adding transform of storage::EdgeAccessor in for loop of query::EdgeAccessor